packages/core/src/browser/window/default-window-service.ts

```diff
@@ -53,14 +53,13 @@ export class DefaultWindowService implements WindowService, FrontendApplicationC
         if (confirmExit === 'never') {
             return true;
         }
+        let preventUnload = false;
         for (const contribution of this.contributions.getContributions()) {
-            if (contribution.onWillStop) {
-                if (!!contribution.onWillStop(this.frontendApplication)) {
-                    return false;
-                }
+            if (contribution.onWillStop?.(this.frontendApplication)) {
+                preventUnload = true;
             }
         }
-        return confirmExit !== 'always';
+        return !preventUnload && confirmExit !== 'always';
```

Comment:
Keeping track of what booleans mean when they're returned negated can be a confusing business.

Maybe:
```typescript
canUnload = confirmExit !== 'always';
// ...loop through and set `canUnload` to false if an `onWillStop` returns truthy
return canUnload;
```
Would be nice if we could rename the `FrontendApplicationContribution` method `safeToUnload` and return `false` rather than `true` or something, but alas, it's too late for that :)
Again, minor, since the current code is consistent with `master`.

Reply:
Not sure I can do better than this indeed :)